app/_components/PoiCard.tsx

```diff
@@ -37,27 +37,34 @@ export function PoiCard({
     latitude: search_latitude,
     longitude: search_longitude,
   };
-
+  //hint useStates
+  const [isOpen, setIsOpen] = useState(false);
+  const [hints, setHints] = useState<string[]>([]);
   // HANDLERS FUNCTIONS
   const handleCheckUserInSearchZone = (): boolean => {
     if (!userCoordinates) return false;
     return (
-      GetDistanceFromCoordinatesToMeters(userCoordinates, pinCoordinates) < payload.search_radius
+      GetDistanceFromCoordinatesToMeters(userCoordinates, pinCoordinates) <
+      payload.search_radius
     );
   };
 
-  const PostGuess = async (user: User, pin: Pin, distance: number):Promise<Response|void> => {
+  const PostGuess = async (
+    user: User,
+    pin: Pin,
+    distance: number
+  ): Promise<Response | void> => {
     try {
-    if (!user) throw 'Not logged in'; //error
-    if (!pin) throw 'Can not get pin';
-
-    const uid = user.uid;
-    const {poi_id, search_radius} = pin;
-    const data: {
-      distance: number;
-      poi_id: number | undefined;
-      uid: string;
-      search_radius: number | undefined;
+      if (!user) throw "Not logged in"; //error
+      if (!pin) throw "Can not get pin";
```

Comment:
Throwing errors here is good, but currently the `catch` block below is just logging the error to the console.

For refactoring this in the future, it might be good to handle the errors in some way (for example, returning an object of `error` string(s) that can be displayed on the frontend to provide feedback to the user.)